### PR TITLE
modules: hal_nordic: nrf_802154 serialization race condition fix

### PR DESCRIPTION
**Pull request by ahasztag**
This fixes a problem with a race condition on nRF53

### CODE REVIEW

File: modules/hal_nordic/nrf_802154/serialization/platform/nrf_802154_spinel_response_notifier.c [hunk 1]
```diff
@@ -10,7 +10,7 @@
 #include <string.h>
 
 #include <zephyr/logging/log.h>
-#include <zephyr/kernel.h>
+#include <zephyr/zephyr.h>
```
Comment (nordicjm) [inline modules/hal_nordic/nrf_802154/serialization/platform/nrf_802154_spinel_response_notifier.c:13]: Wrong change

File: modules/hal_nordic/nrf_802154/serialization/platform/nrf_802154_spinel_response_notifier.c [hunk 2]
```diff
@@ -86,9 +85,6 @@ nrf_802154_spinel_notify_buff_t *nrf_802154_spinel_response_notifier_property_aw
 		LOG_ERR("No response within timeout %u", timeout);
 	}
 
-	ret = k_mutex_unlock(&await_mutex);
-	assert(ret == 0);
-	(void)ret;
 	LOG_DBG("Unlocking response notifier");
```
Comment (jciupis) [inline modules/hal_nordic/nrf_802154/serialization/platform/nrf_802154_spinel_response_notifier.c:88]: This log is misplaced now.

**Comment by jciupis** [comment]
@ahasztag compliance check fails because of empty commit body

File: modules/hal_nordic/nrf_802154/serialization/platform/nrf_802154_spinel_response_notifier.c (hunk 2 above)
Comment (ahasztag) [reply modules/hal_nordic/nrf_802154/serialization/platform/nrf_802154_spinel_response_notifier.c]: Fixed

**Comment by ahasztag** [comment]
> @ahasztag compliance check fails because of empty commit message body

Fixed

**Review by jciupis — APPROVED** [review]

**Review by ankuns — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Pull request merged**